Update SPI device and PLIC IDs

jwnrt opened the pull request:
This PR updates the SPI device with its new registers and interrupts and removes generic mode (no longer used by EG or DJ).

The PLIC interrupt IDs are also updated for all blocks.

jwnrt committed:
[ot] hw/riscv: ot_plic: update per-block interrupt IDs

rivos-eblot commented on hw/opentitan/ot_hmac.c:
```diff
@@ -762,12 +762,15 @@ static void ot_hmac_realize(DeviceState *dev, Error **errp)
     memset(s->ctx, 0, sizeof(*(s->ctx)));
     memset(s->regs, 0, sizeof(*(s->regs)));
 
+    r->cfg = 0x4100;
```
Please favor an unsigned constant (`u` suffix)

rivos-eblot replied:
(or even better use the REG32 macro if you want to, this is something we failed to do up to now :-()

jwnrt committed:
[ot] hw/opentitan: ot_spi_device: update interrupts, regs, remove generic mode

Updates interrupts and registers to latest versions and removes generic
mode entirely.

jwnrt committed:
[ot] hw/opentitan: ot_uart: update interrupts

`TX_EMPTY` has changed to `TX_DONE` which indicates the buffer is empty
and there's no ongoing transmission. Added the interrupt to the module
but not integrated it yet.

jwnrt committed:
[ot] .github/workflows: disable Darjeeling build and test

We have broken Darjeeling functionally with changes to Earl Grey. We
will have to resolve the errors later, but for now it's not the focus.

rivos-eblot commented on .github/workflows/build_test.yaml:
```diff
@@ -147,11 +147,11 @@ jobs:
-          timeout -s KILL 4 ./qemu-system-riscv32 -M ot-darjeeling,no_epmp_cfg=true -nographic \
-            -global ot-ibex_wrapper-dj.lc-ignore=on \
-            -object ot-rom_img,id=rom0,file=exit_dj.bin -d in_asm,int
+      # - name: Check Darjeeling VM execution
```
Please remove these lines for now (avoid dead "code"). Thanks

rivos-eblot commented on hw/opentitan/ot_spi_device.c:
```diff
@@ -52,75 +52,45 @@
-SHARED_FIELD(INTR_READBUF_WATERMARK, 9u, 1u)
-SHARED_FIELD(INTR_READBUF_FLIP, 10u, 1u)
-SHARED_FIELD(INTR_TPM_HEADER_NOT_EMPTY, 11u, 1u)
+SHARED_FIELD(INTR_UPLOAD_CMDFIFO_NOT_EMPTY, 0u, 1u)
```
I just noticed that the original version has been trashed, likely by `clang-format`.

Can you indent (4 space char) the `FIELD` and `SHARED_FIELD` lines, and add the missing `/* clang-format off */` comment before the first register definition and `/* clang-format on */` after the last one? Thanks.

rivos-eblot commented on hw/opentitan/ot_spi_device.c:
```diff
@@ -1840,45 +1593,19 @@ static void ot_spi_device_spi_regs_write(void *opaque, hwaddr addr,
-                              __func__);
-            }
-            break;
+        case CTRL_MODE_DISABLED:
```
I would expected a `LOG_GUEST_ERROR` here (?)

jwnrt replied:
I'm going to `LOG_UNIMP` until disabling is properly implemented.

rivos-eblot commented on hw/opentitan/ot_spi_device.c:
```diff
@@ -2206,9 +1880,6 @@ static void ot_spi_device_chr_handle_header(OtSPIDeviceState *s)
 
     bus->rev_rx = (bool)(mode & R_CFG_RX_ORDER_MASK);
     bus->rev_tx = (bool)(mode & R_CFG_TX_ORDER_MASK);
-    /* if phase or polarity does not match, corrupt data */
```
I think the incoming packet mode still need to be checked, as if the remote end sends invalid SPI mode (phase/polarity) there is no chance for the communication to succeed.

jwnrt replied:
Our SPI_DEVICE now only supports mode 0. Let me know if I've done the check correctly.

rivos-eblot replied:
I think so, but I'm really bad at this exercise (as figuring it out the byte order in crypto things) :)

jwnrt replied:
Oh definitely. We spent a long time going in circles in the office over the HMAC byte ordering code in the other PR.

Thanks for checking

rivos-eblot commented on hw/opentitan/ot_uart.c:
```diff
@@ -47,12 +47,13 @@
     SHARED_FIELD(INTR_TX_WATERMARK, 0u, 1u)
     SHARED_FIELD(INTR_RX_WATERMARK, 1u, 1u)
-    SHARED_FIELD(INTR_TX_EMPTY, 2u, 1u)
+    SHARED_FIELD(INTR_TX_DONE, 2u, 1u)
```
I think you need to add a minimal support for this, as a user would end up having code that does not work and no clue about the failure. This is especially senstive since for some reason the RTL designers have elected to break the compatibilty by relocating an existing bit :-(

If this interrupt is enabled (`INTR_ENABLE`), a `LOG_UNIMP` should be emitted till it is actually supported in an upcoming MR.

jwnrt replied:
Sure, it's the same as `TX_EMPTY` within QEMU anyway, I was just being lazy.

rivos-eblot commented:
Coud you add the component the change applies to in the commit message, such as `[ot] hw/opentitan: ot_uart: <msg>`. Thanks

rivos-eblot reviewed (approved):
LGTM
